apps/coscrad-frontend/src/components/higher-order-components/carousel.tsx

```diff
@@ -0,0 +1,48 @@
+import { Button, Card, CardContent } from '@mui/material';
+import { useState } from 'react';
+import { FunctionalComponent } from '../../utils/types/functional-component';
+
+import { cyclicDecrement, cyclicIncrement } from '../../utils/math';
+
+const STARTING_INDEX = 0;
+
+interface CarouselProps<T> {
+    propsForItems: T[];
+    Presenter: FunctionalComponent<T>;
+}
+
+// TODO We may want a unit test of this component
```

Comment:
We should at least test this behaviour in the children components, if not.

Reply:
[Here](https://www.pivotaltracker.com/story/show/183878408) is a story for that.